airflow/providers/google/cloud/hooks/vertex_ai/custom_job.py

```diff
@@ -0,0 +1,2379 @@
+        :type training_pipeline: str
+        :param retry: Designation of what errors, if any, should be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The timeout for this request.
+        :type timeout: float
+        :param metadata: Strings which should be sent along with the request as metadata.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_pipeline_service_client(region)
+        name = client.training_pipeline_path(project_id, region, training_pipeline)
+
+        client.cancel_training_pipeline(
+            request={
+                'name': name,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def cancel_custom_job(
+        self,
+        project_id: str,
+        region: str,
+        custom_job: str,
+        retry: Retry,
+        timeout: float,
+        metadata: Sequence[Tuple[str, str]],
+    ) -> None:
+        """
+        Cancels a CustomJob. Starts asynchronous cancellation on the CustomJob. The server makes a best effort
+        to cancel the job, but success is not guaranteed. Clients can use
+        [JobService.GetCustomJob][google.cloud.aiplatform.v1.JobService.GetCustomJob] or other methods to
+        check whether the cancellation succeeded or whether the job completed despite cancellation. On
+        successful cancellation, the CustomJob is not deleted; instead it becomes a job with a
+        [CustomJob.error][google.cloud.aiplatform.v1.CustomJob.error] value with a
+        [google.rpc.Status.code][google.rpc.Status.code] of 1, corresponding to ``Code.CANCELLED``, and
+        [CustomJob.state][google.cloud.aiplatform.v1.CustomJob.state] is set to ``CANCELLED``.
+
+        :param project_id: Required. The ID of the Google Cloud project that the service belongs to.
+        :type project_id: str
+        :param region: Required. The ID of the Google Cloud region that the service belongs to.
+        :type region: str
+        :param custom_job: Required. The name of the CustomJob to cancel.
+        :type custom_job: str
+        :param retry: Designation of what errors, if any, should be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The timeout for this request.
+        :type timeout: float
+        :param metadata: Strings which should be sent along with the request as metadata.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_job_service_client(region)
+        name = JobServiceClient.custom_job_path(project_id, region, custom_job)
+
+        client.cancel_custom_job(
+            request={
+                'name': name,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def create_pipeline_job(
```

Comment:
Do we need this? I don't see that we using it.

Reply:
I have answered to the same question in message above

Reply:
👍